src/components/selection-list/list-navigation.ts

```diff
@@ -0,0 +1,44 @@
+import keycode = require('keycode');
+import {clamp} from '../../utils';
+import {getFocusableItemValues, ItemValue} from './basic-list';
+
+export class ListNavigation {
+    private values: ItemValue[] = [];
+    private position: number = -1;
+
+    constructor(listItems: React.ReactNode[]) {
+        this.values = getFocusableItemValues(listItems);
+    }
+
+    public getFocusedValue() {
+        return this.values[this.position];
+    }
+
+    public focusValue(value: ItemValue | undefined) {
+        this.position = (value === undefined) ? -1 : this.values.indexOf(value);
+    }
+
+    public focusIndex(index: number) {
+        this.position = clamp(index, 0, this.values.length - 1);
+    }
+
+    public blur() {
+        this.position = -1;
+    }
+
+    public focusPrevious() {
+        this.position === -1 ? this.focusLast() : this.focusIndex(this.position - 1);
```

Comment:
if position is 0, focusPrevious sets the position to 0. Is this on purpose?
The same is true for focusNext